src/common/src/catalog/column.rs

```diff
@@ -54,6 +55,26 @@ impl From<ColumnId> for i32 {
     }
 }
 
+impl From<&ColumnId> for i32 {
+    fn from(id: &ColumnId) -> i32 {
+        id.0
+    }
+}
+
+impl From<ProstColumnIndex> for ColumnId {
+    fn from(column_index: ProstColumnIndex) -> Self {
+        Self::new(column_index.index as _)
+    }
+}
+
+impl From<ColumnId> for ProstColumnIndex {
+    fn from(column_id: ColumnId) -> Self {
+        Self {
+            index: column_id.0 as _,
+        }
+    }
+}
```

Comment:
looks like a dangerous behavior...

Reply:
+1. We should not confuse ID with index, in favor of table alternating in the future. 🤔